gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ListMapper.java

```diff
@@ -0,0 +1,155 @@
+package uk.nhs.adaptors.pss.translator.mapper;
+
+import java.util.List;
+
+import org.hl7.fhir.dstu3.model.CodeableConcept;
+import org.hl7.fhir.dstu3.model.Coding;
+import org.hl7.fhir.dstu3.model.DateTimeType;
+import org.hl7.fhir.dstu3.model.Encounter;
+import org.hl7.fhir.dstu3.model.ListResource;
+import org.hl7.fhir.dstu3.model.ListResource.ListMode;
+import org.hl7.fhir.dstu3.model.ListResource.ListStatus;
+import org.hl7.fhir.dstu3.model.Meta;
+import org.hl7.fhir.dstu3.model.Period;
+import org.hl7.fhir.dstu3.model.Reference;
+import org.hl7.v3.RCMRMT030101UK04CompoundStatement;
+import org.hl7.v3.RCMRMT030101UK04EhrExtract;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+
+import lombok.RequiredArgsConstructor;
+import uk.nhs.adaptors.pss.translator.service.IdGeneratorService;
+import uk.nhs.adaptors.pss.translator.util.DateFormatUtil;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class ListMapper {
```

Comment:
`ListMapper` name seems generic (as it doesn't say what kind of list it can be used for). If it's only for one kind of ListResource, then maybe `ConsultationListMapper` would be a better name? But if you think we'll later add methods to map other lists, then it's fine :)